Libraries/DirectMLX.h

```diff
@@ -3470,9 +3475,45 @@ namespace dml
         return out;
     }
 
-    // 
-    // TODO: NonZeroCoordinates
-    // 
+    struct NonZeroCoordinatesOutputs
+    {
+        Expression count;
+        Expression coordinates;
+    };
+    inline NonZeroCoordinatesOutputs NonZeroCoordinates(Expression input)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        const auto& inputTensorSizes = inputTensor.sizes;
+        uint32_t dimensionCount = static_cast<uint32_t>(inputTensorSizes.size());
+        TensorDimensions outputSizes = {};
+        uint32_t effectiveRank = -1;
+        uint32_t totalElements = 1;
+        for(uint32_t i = 0; i < dimensionCount; ++i) {
```

Comment:
[nits of style consistency within file] `for (`, `}` on new line, blank line after `struct`.

Reply:
Please resolve after changes made. Unresolving (there's still a { after `if` and no space between `if (`).

Reply:
Changes made !

Reply:
Great : D.